extensions/amp-story/1.0/test/test-amp-story-bookend.js

```diff
@@ -1121,6 +1186,89 @@ describes.realWin('amp-story-bookend', {amp: true}, env => {
     );
   });
 
+  it('should resolve absolute url correctly', async () => {
+    const component = {
+      url: '//othersite.com/other-article.html',
+      domainName: 'example.com',
+      type: 'small',
+      title: 'This is an example landscape article',
+      image: '../../assets/01-iconic-american-destinations.jpg',
+    };
+
+    storyUtils.getDocLocation.restore();
+    const fakeDoc = {
```

Comment:
Can you try using one of these methods to mock the `location` object? I've found three ways to do it that'd yield easier code and much better test coverage. By order of personal preference:

1. https://github.com/ampproject/amphtml/blob/244d6476cae989c9e3219e9a9c5a637ab6914bac/extensions/amp-install-serviceworker/0.1/test/test-amp-install-serviceworker.js#L553-L555

2. https://github.com/ampproject/amphtml/blob/244d6476cae989c9e3219e9a9c5a637ab6914bac/extensions/amp-analytics/0.1/test/test-cookie-writer.js#L273

3. https://github.com/ampproject/amphtml/blob/244d6476cae989c9e3219e9a9c5a637ab6914bac/extensions/amp-access-laterpay/0.1/test/test-amp-access-laterpay.js#L25

Reply:
Whenever we get the location in the bookend, we do it by calling `document.location`. Mocking the `document` object properties in the tests is not as straightforward since it is an iframe inside a shadow dom created by Sinon. 

We could use the window object instead of the document in the bookend code, do you prefer it this way?